Test Error in Build in SystemIdTest.testNostId

**Describe the bug**

```
[ERROR] Failures: 
[ERROR]   SystemIdTest.testNostId:38 
Expecting actual:
  "fe80:0:0:0:aede:48ff:fe00:1122%en4"
to match pattern:
  "\d+\.\d+\.\d+\.\d+"
[INFO] 
[ERROR] Tests run: 100, Failures: 1, Errors: 0, Skipped: 3
```

**To Reproduce**
Steps to reproduce the behavior:
1. On macOS - run `mvn package`
2. Error happens in `nb-api` tests

**What was Expected**
Expect test to always pass.

--- a/nb-api/src/main/java/io/nosqlbench/api/metadata/SystemId.java
+++ b/nb-api/src/main/java/io/nosqlbench/api/metadata/SystemId.java
@@ -28,6 +28,7 @@ import java.nio.charset.StandardCharsets;
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
 import java.util.*;
+import java.util.stream.*;
 
 public class SystemId {
 
@@ -60,6 +61,7 @@ public class SystemId {
                 if (o2.getName().startsWith("e")) return 1;
                 return 0;
             }).findFirst();
+
         if (first.isEmpty()) return Optional.empty();
 
         Optional<InetAddress> firstInetAddrForInterface = first.get().getInterfaceAddresses().stream()
@@ -70,9 +72,21 @@ public class SystemId {
                 if (i2 instanceof Inet4Address) return 1;
                 return 0;
             }).findFirst();
-        return firstInetAddrForInterface;
-    }
 
+        if (firstInetAddrForInterface.orElse(null).getHostAddress().contains(".")) return firstInetAddrForInterface;
+
+        Optional<InetAddress> firstIpAddress = ifaces.stream()
+            .flatMap(i4 -> {
+                Iterator<InetAddress> iterator = i4.getInetAddresses().asIterator();
+                return StreamSupport.stream(
+                                            Spliterators.spliteratorUnknownSize(iterator, Spliterator.ORDERED),
+                                            false
+                                            );
+            }).filter(j4 -> j4.getHostAddress().contains("."))
+            .findFirst();
+
+        return firstIpAddress;
+    }
 
     /**
      * Using this to bypass OSHI because it calls logger init before we want it.

--- a/nb-api/src/main/java/io/nosqlbench/api/metadata/SystemId.java
+++ b/nb-api/src/main/java/io/nosqlbench/api/metadata/SystemId.java
@@ -73,7 +73,8 @@ public class SystemId {
                 return 0;
             }).findFirst();
 
-        if (firstInetAddrForInterface.orElse(null).getHostAddress().contains(".")) return firstInetAddrForInterface;
+        if ( firstInetAddrForInterface.isPresent() &&
+             firstInetAddrForInterface.orElse(null).getHostAddress().contains(".")) return firstInetAddrForInterface;
 
         Optional<InetAddress> firstIpAddress = ifaces.stream()
             .flatMap(i4 -> {

